### PR TITLE
Pds/post az fixes 2 fns

### PR DESCRIPTION
**Pull request by papadeltasierra**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 [hunk 1]
```diff
@@ -78,6 +78,33 @@ INPUTOBJECT <IConnectedCluster>:
...
+function Merge-MaybeNullInput {
+    [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.DoNotExportAttribute()]
+    param(
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.IConnectedCluster]
+        $InputObject,
+        [System.Collections.Generic.Dictionary[system.String, System.Object]]
+        $LclPSBoundParameters
+    )
+
+    $mergeFields = @(
```
Comment (xw-zhang24) [inline src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:90]: Maybe worth a comment explain these are optional fields and we'll read value directly if provided

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 [hunk 2]
```diff
@@ -421,36 +448,43 @@ function Set-AzConnectedKubernetes {
...
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('Infrastructure')) -and $InputObject.PSObject.Properties['Infrastructure']) {
-                $PSBoundParameters.Add('Infrastructure', $InputObject.Infrastructure)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('PrivateLinkState')) -and $InputObject.PSObject.Properties['PrivateLinkState']) {
-                $PSBoundParameters.Add('PrivateLinkState', $InputObject.PrivateLinkState)
-            }
+            Set-PSDebug 2
```
Comment (xw-zhang24) [inline src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:451]: Don't forget to remove these debugging helpers!

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 [hunk 3]
```diff
@@ -421,36 +448,43 @@ function Set-AzConnectedKubernetes {
...
+
+            # Merge the fields that use a common merging process.
+            Merge-MaybeNullInput -InputObject $InputObject -LclPSBoundParameters $PSBoundParameters
+
+            Set-PSDebug 0
+
+            # if ((-not $PSBoundParameters.ContainsKey('WorkloadIdentityEnabled')) -and $InputObject.PSObject.Properties['WorkloadIdentityEnabled']) {
+            #     $WorkloadIdentityEnabled = $InputObject.WorkloadIdentityEnabled
```
Comment (xw-zhang24) [inline src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:459]: `WorkloadIdentityEnabled` and `OidcIssuerProfileEnabled` below will be used by code outside the inputOject parsing block, so even after calling `Merge-MaybeNullInput`, we'll still need to assign values. 

Unfortunately no easy way to get rid of the ugly parsing ;(

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 (hunk 1 above)
Comment (papadeltasierra) [reply src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:90]: Done.

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 (hunk 2 above)
Comment (papadeltasierra) [reply src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:451]: Done.

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 (hunk 3 above)
Comment (papadeltasierra) [reply src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:459]: Where are these used?  They are checked in PSBoundParameters but won't that be OK?

Comment (xw-zhang24) [reply src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:459]: Actually you're right, they are checked in PSBoundParameters https://github.com/xw-zhang24/azure-powershell/blob/generation/src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1#L829

File: src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1 [hunk 4]
```diff
@@ -78,6 +78,25 @@ INPUTOBJECT <IConnectedCluster>:
...
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.IConnectedCluster]
+        $InputObject,
+        [System.Collections.Generic.Dictionary[system.String, System.Object]]
+        $LclPSBoundParameters
+    )
+
+    $mergeFields = 'WorkloadIdentityEnabled', 'OidcIssuerProfileEnabled', 'OidcIssuerProfileSelfHostedIssuerUrl', 'Distribution', 'DistributionVersion', 'Infrastructure', 'PrivateLinkState'
+
+    foreach ($mergeField in $mergeFields) {
+        if ((-not $LclPSBoundParameters.ContainsKey($mergeField)) -and $InputObject.PSObject.Properties[$mergeField] -and $null -ne $InputObject.PSObject.Properties[$mergeField]) {
```
Comment (xw-zhang24) [inline src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1:93]: The last one shall be `$null -ne $InputObject.PSObject.Properties[$mergeField].Value` instead

**Comment by papadeltasierra** [comment]
Superceeded by more changes.

**Pull request closed without merging**